cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -155,8 +155,9 @@ private CloseableIterator<StructuredRow> getFilterByRangeIterator(Range keyRange
   }
 
   private Range getLongestPrefixRange(Range range) {
-    fieldValidator.validatePartialPrimaryKeys(range.getBegin());
-    fieldValidator.validatePartialPrimaryKeys(range.getEnd());
+    // Allow passing in non-primary key range
+    fieldValidator.validateFields(range.getBegin());
```

Comment:
Please ensure range has at least 1 primary key. Otherwise we end up "read all table and do postfiltering" that is not acceptable.

Reply:
But does it matter? The current table can pass in Range.all() that scans the whole table

Reply:
I think that yes in implementation we should try to improve the efficiency but should not limit what caller wants. For example(might not be a very proper one), if we have a profile table and primary key is profile ID. We should allow it if users wants all profiles created last week, though it might be slow.

Reply:
It should. We are not building a general-purpose database. And I can't think of any reasonable use case with whole table sequence scan + post filter. And with new API it's very easy to get that. Previously we did not have automatic postfiltering, so it was pretty obvious if you read the whole table. Now it's really hidden and onobvious. We must prevent shooting yourself in the foot

Reply:
thanks, updated